# Teleop Basic Param

navigation:
  xy_control_mode: 0
  # World Pos Control Mode: 0
  # World Vel Control Mode: 2
  # Local Vel Control Mode: 3
  # Attitude Control Mode: 4

  takeoff_height: 0.8 #1.0
  outdoor_takeoff_height: 1.5

  # teleop operation
  max_target_vel: 0.5
  joy_target_vel_interval: 0.005  # 0.2 / 20 = 0.01, 0.005 ~ 0.01  m/s
  joy_target_z_interval: 0.02
  max_target_yaw_rate: 0.1 #  0.05
  teleop_local_frame: fc

  cmd_vel_lev2_gain : 2.0
  nav_vel_limit : 0.2
  vel_nav_threshold : 0.05

  gain_tunning_mode: 0
  max_target_tilt_angle: 0.2
  cmd_angle_lev2_gain : 1.5

  # gps waypoint
  gps_waypoint_threshold: 3.0
  gps_waypoint_check_du: 1.0

  # desire tilt
  baselink_rot_pub_interval: 0.020
  baselink_rot_change_thresh: 0.01 # 0.2 hz -> 0.04; 0.01 hz-> 0.02
  baselink_rot_angvel: 0.1755

  # transformation planning
  joint_angvel: 0.2  # rad/s actual value is 0.38. smaller is more safe

  # rolling mode
  rolling_max_pitch_ang_vel: 1.0
  rolling_max_yaw_ang_vel: 1.0
  ground_trajectory_duration: 5.0       # should not to be 0
  standing_baselink_roll_converged_thresh: 0.2
  rolling_pitch_update_thresh: 0.2

  # down mode
  down_mode_roll_angvel: 0.2

  # param for joy stick control
  joy_stick_deadzone: 0.2

  # aerial trajectory
  trajectory_mean_vel: 0.1
  trajectory_mean_yaw_rate: 0.1
  trajectory_min_du: 6.0
